src/analyze/analyze.c

```diff
@@ -221,6 +222,35 @@ static UnitTimes* unit_times_free_array(UnitTimes *t) {
 }
 DEFINE_TRIVIAL_CLEANUP_FUNC(UnitTimes*, unit_times_free_array);
 
+static char *rmrdir_and_free(char *p) {
```

Comment:
There's already an rm -rf helper in src/shared/rm-rf.h - can you use that instead?